File: LeetCode/Easy/Transform Array by Parity/3467.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var transformArray = function(nums) {
    let count = 0;
    
    for (num of nums) {
        if (num % 2 === 0) {
            count++;
        }
    }

    for (let j = 0; j < nums.length; j++) {
        nums[j] = (j < count) ? 0 : 1;
    }

    return nums;
};

File: LeetCode/Easy/Transform Array by Parity/3467.java
class Solution {
    public int[] transformArray(int[] nums) {
        int count = 0;
        
        for (int num : nums) {
            if (num % 2 == 0) {
                count++;
            }
        }

        for (int j = 0; j < nums.length; j++) {
            nums[j] = (j < count) ? 0 : 1;
        }

        return nums;
    }
}
